test/e2e/network_policy.go

```diff
@@ -0,0 +1,438 @@
+			Suffix("summary").
+			DoRaw()
+	}
+
+	// The netmonitor container will continuously poll for named service endpoints and then
+	// check connectivity between the two specified services.
+	timeout := time.Now().Add(convergenceTimeout * time.Minute)
+	for i := 0; !passed && timeout.After(time.Now()); i++ {
+		time.Sleep(2 * time.Second)
+		framework.Logf("About to make a proxy summary call")
+		start := time.Now()
+		body, err = getSummary()
+		framework.Logf("Proxy summary call returned in %v", time.Since(start))
+		if err != nil {
+			framework.Logf("Attempt %v: service/pod still starting. (error: '%v')", i, err)
+			continue
+		}
+
+		var summary Summary
+		err = json.Unmarshal(body, &summary)
+		if err != nil {
+			framework.Logf("Warning: unable to unmarshal response (%v): '%v'", string(body), err)
+			continue
+		}
+
+		framework.Logf("Summary: %v", string(body))
+		passed = summary == expected
+		if passed {
+			break
+		}
+	}
+
+	if !passed {
+		if body, err = getDetails(); err != nil {
+			framework.Failf("Timed out. Cleaning up. Error reading details: %v", err)
+		} else {
+			framework.Failf("Timed out. Cleaning up. Details:\n%s", string(body))
+		}
+	}
+	Expect(passed).To(Equal(true))
+}
+
+// Configure namespace network isolation by setting the network-isolation annotation
+// on the namespace.
+func setNetworkIsolationAnnotations(f *framework.Framework, namespace *api.Namespace, enableIsolation bool) {
+	var annotations = map[string]string{}
+	if enableIsolation {
+		By(fmt.Sprintf("Enabling isolation through namespace annotations on namespace %v", namespace.Name))
+		annotations["net.beta.kubernetes.io/network-policy"] = `{"ingress":{"isolation":"DefaultDeny"}}`
+	} else {
+		By(fmt.Sprintf("Disabling isolation through namespace annotations on namespace %v", namespace.Name))
+		delete(annotations, "net.beta.kubernetes.io/network-policy")
+	}
+
+	// Update the namespace.  We set the resource version to be an empty
+	// string, this forces the update.  If we weren't to do this, we would
+	// either need to re-query the namespace, or update the namespace
+	// references with the one returned by the update.  This approach
+	// requires less plumbing.
+	namespace.ObjectMeta.Annotations = annotations
+	namespace.ObjectMeta.ResourceVersion = ""
+	_, err := f.Client.Namespaces().Update(namespace)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+// Add a network policy object to open up ingress traffic to a specific port on a namespace.
+func addNetworkPolicyOpenPort(f *framework.Framework, namespace *api.Namespace, name string, port int32, protocol api.Protocol) {
+	By(fmt.Sprintf("Setting network policy to allow proxy traffic for namespace %v", namespace.Name))
+
+	lport := intstr.IntOrString{IntVal: port}
+
+	framework.Logf("Creating policy %v", name)
+	_, err := f.Client.NetworkPolicies(namespace.Name).Create(&extensions.NetworkPolicy{
+		ObjectMeta: api.ObjectMeta{
+			Name: name,
+		},
+		Spec: extensions.NetworkPolicySpec{
```

Comment:
By adding a `podSelector` here in the Spec, we can open ports on a specific service rather than on all pods within the namespace.